pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -592,6 +594,12 @@ func FallbackGeneratorNameIfNecessary(
 
 			return DeploymentBasicV1Beta1GeneratorName
 		}
+	case DaemonsetAppsV1Beta2GeneratorName:
+		if !Contains(resourcesList, batchv2alpha1.SchemeGroupVersion.WithResource("daemonsets")) {
```

Comment:
batchv2alpha1 does not have "daemonset" resource. appsv1beta2 should be used instead.

Reply:
+1. Agree. Since this is falling back to older version, newest version `appsv1beta2` should be used here.